### PR TITLE
8340082: Use inline return tag in java.base

### PR DESCRIPTION
**Pull request by jddarcy**
Candidates for this refactoring were found programmatically; the program to find candidates is in a comment on the bug.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340082](https://bugs.openjdk.org/browse/JDK-8340082): Use inline return tag in java.base (**Enhancement** - P4)

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Pavel Rappo](https://openjdk.org/census#prappo) (@pavelrappo - **Reviewer**)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20981/head:pull/20981` \
`$ git checkout pull/20981`

Update a local copy of the PR: \
`$ git checkout pull/20981` \
`$ git pull https://git.openjdk.org/jdk.git pull/20981/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20981`

View PR using the GUI difftool: \
`$ git pr show -t 20981`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20981.diff">https://git.openjdk.org/jdk/pull/20981.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20981#issuecomment-2347938526)

### CODE REVIEW

**Review by irisclark — APPROVED** [review]

File: src/java.base/share/classes/java/io/ObjectInputFilter.java [hunk 1]
```diff
@@ -1218,8 +1218,7 @@ private static boolean matchesPackage(Class<?> c, String pkg) {
             }
 
             /**
-             * Returns the pattern used to create this filter.
-             * @return the pattern used to create this filter
+             * {@return the pattern used to create this filter}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/io/ObjectInputFilter.java:1221]:
```suggestion
             * {@return the pattern used to create this filter}.
```

File: src/java.base/share/classes/java/io/ObjectInputStream.java [hunk 2]
```diff
@@ -3859,8 +3859,7 @@ private int readUTFChar(StringBuilder sbuf, long utflen)
         }
 
         /**
-         * Returns the number of bytes read from the input stream.
-         * @return the number of bytes read from the input stream
+         * {@return the number of bytes read from the input stream}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/io/ObjectInputStream.java:3862]:
```suggestion
         * {@return the number of bytes read from the input stream}.
```

File: src/java.base/share/classes/java/lang/annotation/Retention.java [hunk 3]
```diff
@@ -45,8 +45,7 @@
 @Target(ElementType.ANNOTATION_TYPE)
 public @interface Retention {
     /**
-     * Returns the retention policy.
-     * @return the retention policy
+     * {@return the retention policy}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/lang/annotation/Retention.java:48]:
```suggestion
     * {@return the retention policy}.
```

File: src/java.base/share/classes/java/nio/charset/MalformedInputException.java [hunk 4]
```diff
@@ -56,16 +56,14 @@ public MalformedInputException(int inputLength) {
     }
 
     /**
-     * Returns the length of the input.
-     * @return the length of the input
+     * {@return the length of the input}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/nio/charset/MalformedInputException.java:59]:
```suggestion
     * {@return the length of the input}.
```

File: src/java.base/share/classes/java/nio/charset/MalformedInputException.java [hunk 5]
```diff
@@ -56,16 +56,14 @@ public MalformedInputException(int inputLength) {
...
      */
     public int getInputLength() {
         return inputLength;
     }
 
     /**
-     * Returns the message.
-     * @return the message
+     * {@return the message}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/nio/charset/MalformedInputException.java:66]:
```suggestion
     * {@return the message}.
```

File: src/java.base/share/classes/java/nio/charset/UnmappableCharacterException.java [hunk 6]
```diff
@@ -56,16 +56,14 @@ public UnmappableCharacterException(int inputLength) {
     }
 
     /**
-     * Returns the length of the input.
-     * @return the length of the input
+     * {@return the length of the input}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/nio/charset/UnmappableCharacterException.java:59]:
```suggestion
     * {@return the length of the input}.
```

File: src/java.base/share/classes/java/nio/charset/UnmappableCharacterException.java [hunk 7]
```diff
@@ -56,16 +56,14 @@ public UnmappableCharacterException(int inputLength) {
...
      */
     public int getInputLength() {
         return inputLength;
     }
 
     /**
-     * Returns the message.
-     * @return the message
+     * {@return the message}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/nio/charset/UnmappableCharacterException.java:66]:
```suggestion
     * {@return the message}.
```

File: src/java.base/share/classes/java/time/format/TextStyle.java [hunk 8]
```diff
@@ -137,8 +137,7 @@ public boolean isStandalone() {
     }
 
     /**
-     * Returns the stand-alone style with the same size.
-     * @return the stand-alone style with the same size
+     * {@return the stand-alone style with the same size}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/time/format/TextStyle.java:140]:
```suggestion
     * {@return the stand-alone style with the same size}.
```

File: src/java.base/share/classes/java/util/concurrent/SynchronousQueue.java [hunk 9]
```diff
@@ -477,8 +477,7 @@ public Spliterator<E> spliterator() {
     }
 
     /**
-     * Returns a zero-length array.
-     * @return a zero-length array
+     * {@return a zero-length array}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/SynchronousQueue.java:480]:
```suggestion
     * {@return a zero-length array}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicBoolean.java [hunk 10]
```diff
@@ -176,8 +176,7 @@ public final boolean getAndSet(boolean newValue) {
     }
 
     /**
-     * Returns the String representation of the current value.
-     * @return the String representation of the current value
+     * {@return the String representation of the current value}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicBoolean.java:179]:
```suggestion
     * {@return the String representation of the current value}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicInteger.java [hunk 11]
```diff
@@ -340,8 +340,7 @@ public final int accumulateAndGet(int x,
     }
 
     /**
-     * Returns the String representation of the current value.
-     * @return the String representation of the current value
+     * {@return the String representation of the current value}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicInteger.java:343]:
```suggestion
     * {@return the String representation of the current value}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicIntegerArray.java [hunk 12]
```diff
@@ -366,8 +366,7 @@ public final int accumulateAndGet(int i, int x,
     }
 
     /**
-     * Returns the String representation of the current values of array.
-     * @return the String representation of the current values of array
+     * {@return the String representation of the current values of array}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicIntegerArray.java:369]:
```suggestion
     * {@return the String representation of the current values of array}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicLong.java [hunk 13]
```diff
@@ -341,8 +341,7 @@ public final long accumulateAndGet(long x,
     }
 
     /**
-     * Returns the String representation of the current value.
-     * @return the String representation of the current value
+     * {@return the String representation of the current value}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicLong.java:344]:
```suggestion
     * {@return the String representation of the current value}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicLongArray.java [hunk 14]
```diff
@@ -366,8 +366,7 @@ public final long accumulateAndGet(int i, long x,
     }
 
     /**
-     * Returns the String representation of the current values of array.
-     * @return the String representation of the current values of array
+     * {@return the String representation of the current values of array}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicLongArray.java:369]:
```suggestion
     * {@return the String representation of the current values of array}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicReference.java [hunk 15]
```diff
@@ -269,8 +269,7 @@ public final V accumulateAndGet(V x,
     }
 
     /**
-     * Returns the String representation of the current value.
-     * @return the String representation of the current value
+     * {@return the String representation of the current value}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicReference.java:272]:
```suggestion
     * {@return the String representation of the current value}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicReferenceArray.java [hunk 16]
```diff
@@ -297,8 +297,7 @@ public final E accumulateAndGet(int i, E x,
     }
 
     /**
-     * Returns the String representation of the current values of array.
-     * @return the String representation of the current values of array
+     * {@return the String representation of the current values of array}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicReferenceArray.java:300]:
```suggestion
     * {@return the String representation of the current values of array}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/DoubleAccumulator.java [hunk 17]
```diff
@@ -189,8 +189,7 @@ public double getThenReset() {
     }
 
     /**
-     * Returns the String representation of the current value.
-     * @return the String representation of the current value
+     * {@return the String representation of the current value}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/DoubleAccumulator.java:192]:
```suggestion
     * {@return the String representation of the current value}.
```

File: src/java.base/share/classes/java/util/concurrent/atomic/LongAccumulator.java [hunk 18]
```diff
@@ -183,8 +183,7 @@ public long getThenReset() {
     }
 
     /**
-     * Returns the String representation of the current value.
-     * @return the String representation of the current value
+     * {@return the String representation of the current value}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/concurrent/atomic/LongAccumulator.java:186]:
```suggestion
     * {@return the String representation of the current value}.
```

File: src/java.base/share/classes/java/util/jar/Manifest.java [hunk 19]
```diff
@@ -119,8 +119,7 @@ public Manifest(Manifest man) {
     }
 
     /**
-     * Returns the main Attributes for the Manifest.
-     * @return the main Attributes for the Manifest
+     * {@return the main Attributes for the Manifest}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/jar/Manifest.java:122]:
```suggestion
     * {@return the main Attributes for the Manifest}.
```

File: src/java.base/share/classes/java/util/zip/Deflater.java [hunk 20]
```diff
@@ -782,8 +782,7 @@ public int deflate(ByteBuffer output, int flush) {
     }
 
     /**
-     * Returns the ADLER-32 value of the uncompressed data.
-     * @return the ADLER-32 value of the uncompressed data
+     * {@return the ADLER-32 value of the uncompressed data}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/zip/Deflater.java:785]:
```suggestion
     * {@return the ADLER-32 value of the uncompressed data}.
```

File: src/java.base/share/classes/java/util/zip/Inflater.java [hunk 21]
```diff
@@ -282,8 +282,7 @@ public boolean needsInput() {
     }
 
     /**
-     * Returns true if a preset dictionary is needed for decompression.
-     * @return true if a preset dictionary is needed for decompression
+     * {@return true if a preset dictionary is needed for decompression}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/zip/Inflater.java:285]:
```suggestion
     * {@return true if a preset dictionary is needed for decompression}.
```

File: src/java.base/share/classes/java/util/zip/Inflater.java [hunk 22]
```diff
@@ -293,10 +292,8 @@ public boolean needsDictionary() {
     }
 
     /**
-     * Returns true if the end of the compressed data stream has been
-     * reached.
-     * @return true if the end of the compressed data stream has been
-     * reached
+     * {@return true if the end of the compressed data stream has been
+     * reached}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/zip/Inflater.java:296]:
```suggestion
     * {@return true if the end of the compressed data stream has been
     * reached}.
```

File: src/java.base/share/classes/java/util/zip/Inflater.java [hunk 23]
```diff
@@ -602,8 +599,7 @@ public int inflate(ByteBuffer output) throws DataFormatException {
     }
 
     /**
-     * Returns the ADLER-32 value of the uncompressed data.
-     * @return the ADLER-32 value of the uncompressed data
+     * {@return the ADLER-32 value of the uncompressed data}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/zip/Inflater.java:602]:
```suggestion
     * {@return the ADLER-32 value of the uncompressed data}.
```

File: src/java.base/share/classes/java/util/zip/ZipEntry.java [hunk 24]
```diff
@@ -138,8 +138,7 @@ public ZipEntry(ZipEntry e) {
     }
 
     /**
-     * Returns the name of the entry.
-     * @return the name of the entry
+     * {@return the name of the entry}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/zip/ZipEntry.java:141]:
```suggestion
     * {@return the name of the entry}.
```

File: src/java.base/share/classes/java/util/zip/ZipFile.java [hunk 25]
```diff
@@ -498,8 +498,7 @@ public int available() throws IOException {
     }
 
     /**
-     * Returns the path name of the ZIP file.
-     * @return the path name of the ZIP file
+     * {@return the path name of the ZIP file}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/zip/ZipFile.java:501]:
```suggestion
     * {@return the path name of the ZIP file}.
```

File: src/java.base/share/classes/java/util/zip/ZipFile.java [hunk 26]
```diff
@@ -560,8 +559,7 @@ public Iterator<T> asIterator() {
     }
 
     /**
-     * Returns an enumeration of the ZIP file entries.
-     * @return an enumeration of the ZIP file entries
+     * {@return an enumeration of the ZIP file entries}
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/util/zip/ZipFile.java:562]:
```suggestion
     * {@return an enumeration of the ZIP file entries}.
```

**Review by ExE-Boss — COMMENTED** [review]
The old version of the doc comments had a `.` at the end of the first sentence:

**Comment by pavelrappo** [comment]
> The old version of the doc comments had a `.` at the end of the first sentence:

The new version has it too, but in the final, generated form: https://docs.oracle.com/en/java/javase/22/docs/specs/javadoc/doc-comment-spec.html#return

`{@return blah}` expands to `Returns blah.`

**Review by pavelrappo — APPROVED** [review]
Thanks, Joe. This looks good, especially considering that the change was produced with the help of a quick-and-dirty, regex-based script. This change is certainly enough to spread awareness of this relatively new javadoc feature.

I remember I had a more involved script that used `javax.lang.model` and string similarity metrics. That script captured a lot more candidates for `{@return}`. But on the other hand, once you start considering non-exact matches, it requires human judgement and increases review effort.

---

Separately, this PR has helped me put a finger on what I __don't__ like about `{@return}`. What I don't like is the generated HTML. `{@return}` saves mental effort when reading raw javadoc in source, but it provides no similar service to the reader of the final, HTML form. Maybe it's just me, but it looks needlessly bloated and silly:

browser screenshot of javax.lang.model.element.AnnotationValue#getValue

Maybe if a method's main description consisted only of `{@return}`, we could skip the first sentence in the "Method Details" section and just output `Returns:`? Any further discussion should happen on the javadoc-dev mailing list.

**Review by LanceAndersen — APPROVED** [review]
I went through the changes and they look fine.

**Comment by liach** [comment]
This patch only captures one-line returns without extra sentences. Is it possible for us to handle slightly more complex documentations like `ParameterizedType::getActualTypeArguments`? https://github.com/openjdk/jdk/blob/f7cb6e23194c8e08a699eca84ae9b424c0283588/src/java.base/share/classes/java/lang/reflect/ParameterizedType.java#L50-L58

@ExE-Boss `{@return abc}` is rendered like `Returns abc.` in the first sentence; note the period addition.

**Comment by pavelrappo** [comment]
> This patch only captures one-line returns without extra sentences. Is it possible for us to handle slightly more complex documentations like `ParameterizedType::getActualTypeArguments`?

Sure, it is possible. However, there's a tradeoff between comprehensiveness and the effort required. Joe's script provides a lot for a little.

If there's interest in it, we could file follow-up bugs to fix more cases with more powerful tools.

**Review by djelinski — APPROVED** [review]

**Review by naotoj — APPROVED** [review]
`java.nio.charset` and `java.time.format` changes look good

**Comment by jddarcy** [comment]
> > This patch only captures one-line returns without extra sentences. Is it possible for us to handle slightly more complex documentations like `ParameterizedType::getActualTypeArguments`?
> 
> Sure, it is possible. However, there's a tradeoff between comprehensiveness and the effort required. Joe's script provides a lot for a little.
> 
> If there's interest in it, we could file follow-up bugs to fix more cases with more powerful tools.

Yes, the quick and dirty program is only a "minimum viable product" level of functionality. It is not complete and doesn't catch all cases. Future improvements welcome!

**Review by liach — APPROVED** [review]
Looks good. It might be feasible to run a more complex tool that analyzes the tokenized javadoc AST from javac as later work.

Annotation changes look good.

**Pull request closed without merging**